test/functional/rpc_signrawtransaction.py

```diff
@@ -45,6 +45,14 @@ def successful_signing_test(self):
         # 2) No script verification error occurred
         assert 'errors' not in rawTxSigned
 
+    def test_with_lock_outputs(self):
+        """Test correct error reporting when trying to sign a locked output"""
+        self.nodes[0].encryptwallet("password")
+        self.restart_node(0)
```

Comment:
Could add a second commit to remove the `restart_node`, which is not required on master, but on 0.17.

Reply:
done

Reply:
just need one more change, the signature method must be changed to be 0.16 compliant

Reply:
now it's properly backported